x-pack/plugins/task_manager/server/MONITORING.md

```diff
@@ -0,0 +1,242 @@
+If you wish to enable DEBUG logging in your Kibana instance, you will need to add the following to your `Kibana.yml`:
+```
+logging:
+  loggers:
+      - context: plugins.taskManager
+        appenders: [console]
+        level: debug
+```
+
+Please bear in mind that these stats are logged as often as your `poll_interval` configuration, which means it could add substantial noise to your logs.
+We would recommend only enabling this level of logging temporarily.
+
+### Understanding the Exposed Stats
+
+As mentioned above, the `health` api exposes three sections: `configuration`, `workload` and `runtime`.
+Each section has a `timestamp` and a `status` which indicates when the last update to this setion took place and whether the health of this section was evaluated as `OK`, `Warning` or `Error`.
+
+The root has its own `status` which indicate the state of the system overall as infered from the `status` of the section.
+An `Error` status in any section will cause the whole system to display as `Error`.
+A `Warning` status in any section will cause the whole system to display as `Warning`.
+An `OK` status will only be displayed when all sections are marked as `OK`.
+
+The root `timestamp` is the time in which the summary was exposed (either to the DEBUG logger or the http api) and the `lastUpdate` is the last time any one of the sections was updated.
+
+#### The Configuration Section
+The `configuration` section summarizes Task Manager's current configuration, including dynamic configurations which change over time, such as `poll_interval` and `max_workers` which adjust in reaction to changing load on the system.
+
+These are "Hot" stats which are updated whenever a change happens in the configuration.
+
+#### The Workload Section
+The `workload` which summarizes the work load in the current deployment, listing the tasks in the system, their types and what their current status is.
+
+It includes three sub sections:
+  - The number of tasks scheduled in the system, broken down by type and status.
+  - The number of idle `overdue` tasks, whose `runAt` has expired.
+  - Execution density in the next minute or so (configurable), which shows how many tasks are scheduled to execute in the scope of each polling interval. This can give us an idea of how much load there is on the current Kibana deployment.
+
+These are "Cold" stat which are updated at a regular cadence, configured by the `monitored_aggregated_stats_refresh_rate` config.
+
+#### The Runtime Section
+The `runtime` tracks Task Manager's performance as it runs, making note of task execution time, _drift_ etc.
+These include:
+  - The time it takes a task to run (mean and median, using a configurable running average window, `50` by default)
+  - The average _drift_ that tasks experience (mean and median, using the same configurable running average window as above). Drift tells us how long after a task's scheduled a task typically executes.
```

Comment:
Would it be more useful to have percentiles instead of mean/median?  P50, P90, P99 seem like a typical set of values, sometimes throw a P95 in there as well.

Reply:
Yeah, I was actually thinking the same, but went with mean/median as a start because we already do that for the perf tests, so it was the easiest first step.

I'll look into how simple it would be to implement that, shouldn't be difficult.